File: afrecog/src/main/java/co/edu/fuac/afrecog/FImage.java
/*
 * Universidad Autónoma de Colombia. 2017. All Rights Reserved.
 * Alexander Cardona
 *
 *
 * This software is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package co.edu.fuac.afrecog;

/*
 * Represents an image or a fragment of it. It is a wrapper for the class with same name into the
 * library
 */
public class FImage {

    private int id;
    private boolean isActive;

    private FImage(){
        // avoid creation without id
    }

    public FImage( int id ){
        this.id  = id;
        isActive = true;
    }

    /* The image is not longer needed */
    public void close(){

        isActive = false;
    }

    public int getWidth(){

        if( isActive ){
            return FImageContext.getInstance().getWidth(this);
        }

        return 0;
    }

    public int getHeight(){

        if( isActive ){
            return FImageContext.getInstance().getHeight(this);
        }

        return 0;
    }

    public int getId(){
        return id;
    }

    /* If image is destroyed, remove image from library */
    @Override
    protected void finalize() throws Throwable {

        if( isActive ){
            FImageContext.getInstance().removeImage(this);
        }

        super.finalize();
    }
}

File: afrecog/src/main/cpp/FImage.cpp
/*
 * Universidad Autónoma de Colombia. 2017. All Rights Reserved.
 * Alexander Cardona
 *
 *
 * This software is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
#include "include/FImage.h"

using namespace uac;

FImage:: FImage(int i, Mat m):
        mat(m), id(i)
{
}

FImage:: ~FImage(){

    //Nothing to do
}

Mat FImage:: getMat(){

    return mat;
}

void FImage:: rotate(int degrees){

    Mat dst;

    Size src_sz = mat.size();
    Size dst_sz(src_sz.height, src_sz.width);

    int len = std::max(mat.cols, mat.rows);
    Point2f center(len/2.0, len/2.0);
    Mat rot_mat = cv::getRotationMatrix2D(center, degrees, 1.0);
    warpAffine(mat, dst, rot_mat, dst_sz);

    mat = dst;
}

